src/test/java/io/github/thebusybiscuit/slimefun4/api/events/TestSlimefunBlockBreakEvent.java

```diff
@@ -0,0 +1,120 @@
+    public void beforeEach() {
+        server.getPluginManager().clearEvents();
+    }
+
+    @Test
+    @DisplayName("Test firing Slimefun Block Break Event")
+    void testEventIsFired() {
+        Player player = new PlayerMock(server, "SomePlayer");
+
+        World world = server.addSimpleWorld("my_world");
+        Block block = new BlockMock(Material.GREEN_TERRACOTTA, new Location(world, 1, 1, 1));
+
+        Slimefun.getRegistry().getWorlds().put("my_world", new BlockStorage(world));
+        BlockStorage.addBlockInfo(block, "id", "FOOD_COMPOSTER");
+
+        server.getPluginManager().callEvent(new BlockBreakEvent(block, player));
+        server.getPluginManager().assertEventFired(SlimefunBlockBreakEvent.class, e -> true);
+    }
+
+    @Test
+    @DisplayName("Test getting the getters from the event")
+    void testGetters() {
+        Player player = new PlayerMock(server, "SomePlayer");
+        ItemStack itemStack = new ItemStack(Material.IRON_PICKAXE);
+        player.getInventory().setItemInMainHand(itemStack);
+
+        World world = server.addSimpleWorld("my_world");
+        Block block = new BlockMock(Material.GREEN_TERRACOTTA, new Location(world, 1, 1, 1));
+
+        Slimefun.getRegistry().getWorlds().put("my_world", new BlockStorage(world));
+        BlockStorage.addBlockInfo(block, "id", "FOOD_COMPOSTER");
+
+        server.getPluginManager().callEvent(new BlockBreakEvent(block, player));
+        server.getPluginManager().assertEventFired(SlimefunBlockBreakEvent.class, e -> {
+            Assertions.assertEquals(block, e.getBlockBroken());
+            Assertions.assertEquals(slimefunItem, e.getSlimefunItem());
+            Assertions.assertEquals(itemStack, e.getHeldItem());
+            Assertions.assertEquals(player, e.getPlayer());
+            Assertions.assertFalse(e.isCancelled());
+            return true;
+        });
+    }
+
+    @Test
+    @DisplayName("Test the event cancelling")
+    void testIsCancelled() {
+        server.getPluginManager().registerEvents(new Listener() {
+            @EventHandler
+            public void onBlockBreak(SlimefunBlockBreakEvent event) {
+                event.setCancelled(true);
+            }
+        }, plugin);
+
+        Player player = new PlayerMock(server, "SomePlayer");
+        ItemStack itemStack = new ItemStack(Material.IRON_PICKAXE);
+        player.getInventory().setItemInMainHand(itemStack);
+
+        World world = server.addSimpleWorld("my_world");
+        Block block = new BlockMock(Material.GREEN_TERRACOTTA, new Location(world, 1, 1, 1));
+
+        Slimefun.getRegistry().getWorlds().put("my_world", new BlockStorage(world));
+        BlockStorage.addBlockInfo(block, "id", "FOOD_COMPOSTER");
+
+        server.getPluginManager().callEvent(new BlockBreakEvent(block, player));
+        server.getPluginManager().assertEventFired(SlimefunBlockBreakEvent.class, e -> {
+            Assertions.assertTrue(e.isCancelled());
+            return true;
+        });
```

Comment:
We cancel the BlockBreakEvent inside SlimefunBlockBreakEvent. Let's assert that is happening.
```suggestion
        BlockBreakEvent blockBreakEvent = new BlockBreakEvent(block, player);
        server.getPluginManager().callEvent(blockBreakEvent);
        server.getPluginManager().assertEventFired(SlimefunBlockBreakEvent.class, e -> {
            Assertions.assertTrue(e.isCancelled());
            Assertions.assertTrue(blockBreakEvent.isCancelled());
            return true;
        });
```